beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -44,22 +44,24 @@ func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockR
 		return nil, err
 	}
 
-	registered, err := vs.validatorRegistered(ctx, altairBlk.ProposerIndex)
-	if registered && err == nil {
-		builderReady, b, err := vs.GetAndBuildBlindBlock(ctx, altairBlk)
-		if err != nil {
-			// In the event of an error, the node should fall back to default execution engine for building block.
-			log.WithError(err).Error("Failed to build a block from external builder, falling " +
-				"back to local execution client")
-			builderGetPayloadMissCount.Inc()
-		} else if builderReady {
-			return b, nil
+	if !req.SkipMevBoost {
+		registered, err := vs.validatorRegistered(ctx, altairBlk.ProposerIndex)
+		if registered && err == nil {
+			builderReady, b, err := vs.GetAndBuildBlindBlock(ctx, altairBlk)
+			if err != nil {
+				// In the event of an error, the node should fall back to default execution engine for building block.
+				log.WithError(err).Error("Failed to build a block from external builder, falling " +
+					"back to local execution client")
+				builderGetPayloadMissCount.Inc()
+			} else if builderReady {
+				return b, nil
+			}
+		} else if err != nil {
+			log.WithFields(logrus.Fields{
+				"slot":           req.Slot,
+				"validatorIndex": altairBlk.ProposerIndex,
+			}).Errorf("Could not determine validator has registered. Default to local execution client: %v", err)
 		}
-	} else if err != nil {
-		log.WithFields(logrus.Fields{
-			"slot":           req.Slot,
-			"validatorIndex": altairBlk.ProposerIndex,
-		}).Errorf("Could not determine validator has registered. Default to local execution client: %v", err)
```

Comment:
Please add tests for these changes

Reply:
This has been tested. The unit test is `TestServer_GetBellatrixBeaconBlock_BuilderCase`
Keep in mind that `req.SkipMevBoost ` is false by default

Reply:
`req.SkipMevBoost` is true is tested through `TestProduceBlockV2SSZ`  and `TestProduceBlockV2`.
`req.SkipMevBoost` is false is tested through `TestServer_GetBellatrixBeaconBlock_BuilderCase`. Local test case.

Reply:
Also added another local test `TestServer_GetBellatrixBeaconBlock_LocalProgressingWithBuilderSkipped`